lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -4834,11 +4834,6 @@ void ConformanceChecker::ensureRequirementsAreSatisfied() {
   Conformance->finishSignatureConformances();
   auto proto = Conformance->getProtocol();
 
-  if (CheckedRequirementSignature)
```

Comment:
This looks like a recursion breaker. If we can kill it off, I'm glad.

Reply:
Not sure why @DougGregor added it in back in d8c8dd658502, but it looks like we can do without it: `ensureRequirementsAreSatisfied` is only called once all type witnesses are resolved, so it cannot recurse into the same conformance checker.